.ci/run_tests.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+# check for docker-compose and docker availability
+which docker > /dev/null || {
+	echo "Please install docker" >&2
+	exit 1
+}
+which docker-compose > /dev/null || {
+	echo "Please install docker-compose" >&2
+	exit 1
+}
+
+export QT_VERSION=5 # TODO: ask user for this one
+export DISTRO_VERSION=21.10 # TODO: ask user for this one
+# can be: ALL, ALL_BUT_PROVIDERS, POSTGRES, HANA, ORACLE, SQLSERVER
+TESTS_TO_RUN=ALL_BUT_PROVIDERS # TODO: ask user for this one
+
+cd $(dirname $0)/..
+export QGIS_WORKSPACE=`pwd`
+#echo "PWD is $PWD"
+#echo "QGIS_WORKSPACE is $QGIS_WORKSPACE"
+
+test -d build || {
```

Comment:
why is this needed?

I think QGIS needs to be built inside the dependencies image, something like:

```
docker pull qgis/qgis3-build-deps:latest
docker tag qgis/qgis3-build-deps:latest qgis3-build-deps


docker run -t --name qgis_container \
                     -v $(pwd):/root/QGIS \
                     --env-file .docker/docker-variables.env \
                     --env PUSH_TO_CDASH=true \
                     --env WITH_QT5=true \
                     --env WITH_QT6=false \
                     --env WITH_QUICK=false \
                     --env WITH_3D=false \
                     --env PATCH_QT_3D=false \
                     qgis3-build-deps \
                     /root/QGIS/.docker/docker-qgis-build.sh
```

Reply:
You are right, testing in docker requires also BUILDING in docker. Unfortunately both the builder and the test runner at the moment hard-code the `build/` directory which means it would clash with a pre-existing build/ directory. Maybe I'll make that parametrizable. Do you know why .docker/docker-qgis-build.sh uses ccache at all ? How are those caches supposed to be surviving multiple runs ?

Reply:
I guess we can always improve this in a second stage, for now it'd be good enough to be able to run a script to run the same job run by the official CI in the same (dockerized) environment.